### PR TITLE
fix(GlobalHeader): Spacing fixes

### PR DESCRIPTION
**Pull request by katiezutter**
### Overview

Fixes several issues discovered in the desk check for GlobalHeader.

1. Removes left and right padding to first and last elements in an AppBar section, respectively
2. Adds behavior to address a long `displayName`
3. Updates padding on the mobile menu Log In / Sign Up buttons
4. Sets the desktop height to 80px and the mobile menu height to 64px (to match monolith)

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [EGG-681](https://codecademy.atlassian.net/browse/EGG-681)
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@9.3.2-alpha.fc3665.0<br/>@codecademy/styleguide@25.2.2-alpha.fc3665.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6025497206092a29d10985a8--gamut-preview.netlify.app>https://6025497206092a29d10985a8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6025497206092a29d10985a8>Deploy Logs</a></p>

File: packages/gamut-labs/src/experimental/AppHeader/index.tsx [hunk 1]
```diff
@@ -38,62 +38,57 @@ export const mapItemToElement = (
...
 export const AppHeader: React.FC<AppHeaderProps> = ({ action, items }) => {
+  const mapItemsToElement = <T extends AppHeaderItem[]>(items: T) => {
+    return items.map((item, index) => (
+      <Box
+        key={item.id}
+        marginLeft={index === 0 ? 0 : 8}
+        marginRight={index === items.length - 1 ? 0 : 8}
+      >
+        {mapItemToElement(action, item)}
+      </Box>
```
Comment (katiezutter) [inline packages/gamut-labs/src/experimental/AppHeader/index.tsx:81]: Removed `AppHeaderTab` and replaced it here, since it was just a fancy `Box` anyway.

Comment (katiezutter) [reply packages/gamut-labs/src/experimental/AppHeader/index.tsx:81]: And refactored to use the same `mapItemsToElement` function used in the mobile header.

Comment (saghdaey) [reply packages/gamut-labs/src/experimental/AppHeader/index.tsx:81]: i was looking at that like: didn't i do this already? then realized yes, only in mobile header hehe
thanks for updating!

Comment (katiezutter) [reply packages/gamut-labs/src/experimental/AppHeader/index.tsx:81]: Yep just copied your brilliance! ☀️

Comment (saghdaey) [reply packages/gamut-labs/src/experimental/AppHeader/index.tsx:81]: ie aaron's suggestion :)

**Review by JoshuaKGoldberg — APPROVED** [review]
🌮

**Review by dreamwasp — APPROVED** [review]
✨

**Review by rogervera — APPROVED** [review]
LGTM.  When we port the header to the monolith, will we revert the height and width to what it was?

**Comment by katiezutter** [comment]
@rogervera not going to revert - I think we want the 80px desktop / 64px mobile header height on both the monolith and static-sites as an end state!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@9.3.3-alpha.91f2d1.0<br/>@codecademy/styleguide@25.2.3-alpha.91f2d1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60256beb81d6a6540c9af1c9--gamut-preview.netlify.app>https://60256beb81d6a6540c9af1c9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60256beb81d6a6540c9af1c9>Deploy Logs</a></p>

**Pull request merged**